[FLINK-17599][docs] Update documents due to FLIP-84 (#12577)
apache/flink: docs/dev/table/sql/queries.md

---

> title: "SELECT queries"

Reviewer: In this section, Do we need change this? all queries should be SELECT queries from my understanding. Others should belong to other kind of SQL, "query" should not equals "SQL"

Author: change to `SELECT statements` which style is the same as others

> title: "SELECT statements"

Reviewer 2: Remove `Operations` section as we moved them into separate pages.

Author: I will revert `SELECT statements` to `Queries` because Queries contain not only SELECT statement but also VALUES statement

                 .field("amount", DataTypes.BIGINT()))
    .create_temporary_table("RubberOrders")

# run an INSERT SQL on the Table and emit the result to the TableSink
table_env \
    .execute_sql("INSERT INTO RubberOrders SELECT product, amount FROM Orders WHERE product LIKE '%Rubber%'")
{% endhighlight %}
</div>
</div>

{% top %}

## Execute a Query

> A SELECT statement or a VALUES statement can be executed to collect the content to local through the `TableEnvironment.executeSql()` method. The method returns the result of the SELECT statement (or the VALUES statement) as a `TableResult`. Similar to SELECT statement, a `Table` can also be executed to collect to content of the table to local through the `Table.execute()` method.

Reviewer 3: ```suggestion
A SELECT statement or a VALUES statement can be executed to collect the content to local through the `TableEnvironment.executeSql()` method. The method returns the result of the SELECT statement (or the VALUES statement) as a `TableResult`. Similar to a SELECT statement, a `Table` object can be executed using the `Table.execute()` method to collect the content of the query to the local client.
```

`TableResult.collect()` method returns a closeable row iterator. The select job will not be finished unless all result data has been collected. We should actively close the job to avoid resource leak through the `CloseableIterator#close()` method. 

> We can also print the select result to client console through the `TableResult.print()` method. The result data in `TableResult` can be accessed only once, so for a `TableResult` instance, `collect()` method and `print()` method can't be called both.

Reviewer 3: ```suggestion
We can also print the select result to client console through the `TableResult.print()` method. The result data in `TableResult` can be accessed only once. Thus, `collect()` and `print()` must not be called after each other.
```

                 .field("amount", DataTypes.BIGINT()))
    .create_temporary_table("RubberOrders")

# run an INSERT SQL on the Table and emit the result to the TableSink
table_env \
    .execute_sql("INSERT INTO RubberOrders SELECT product, amount FROM Orders WHERE product LIKE '%Rubber%'")
{% endhighlight %}
</div>
</div>

{% top %}

## Execute SELECT statement
A SELECT statement can be executed to collect the content to local through the `TableEnvironment.executeSql()` method. The method returns the result of the SELECT statement as a `TableResult`. Similar to SELECT statement, a `Table` can also be executed to collect to content of the table to local through the `Table.execute()` method.
`TableResult.collect()` method returns a closeable row iterator. The select job will not be finished unless all result data has been collected. We should actively close the job to avoid resource leak through the `CloseableIterator#close()` method. 
We can print the select result to client console through the `TableResult.print()` method. But we should make sure all result data to print should be small, because all data will be collected to local memory first, and then print them to console.

> **Notes:** Only append-only streaming SELECT statement is supported now.

Reviewer 2: ```suggestion
**Notes:** For streaming mode, only append-only query is supported now.
```
